### PR TITLE
feat(correlation): Webhooks now have access to the correlation result…

### LINKED ISSUE #1578: Provide access to correlation information in the response body expression
What should we do?

We should provide the result of successful correlations via the webhook response body expression by making it available in the expression context. We will re-use the existing correlation variable to expose this functionality.

Why should we do it?

Customers are requesting access to it: https://jira.camunda.com/browse/SUPPORT-19621
```[tasklist]
### Tasks
- [ ] https://github.com/camunda/connectors/pull/1584
- [x] Backport PR to 8.3: https://github.com/camunda/connectors/pull/1584#issuecomment-1855843749
- [x] Document the correlation object in the docs for the Webhook Connector: https://github.com/camunda/camunda-docs/pull/3098
```

### PR DESCRIPTION
**Pull request by sbuettner**
… in the response body expression

## Description

Provide access to the correlation result in the response body expression of the webhook connector.

## Related issues

closes #1578

### CODE REVIEW

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java [hunk 1]
```diff
@@ -139,12 +131,58 @@ protected ResponseEntity verify(
       response =
           new ResponseEntity<>(verificationResult.body(), headers, verificationResult.statusCode());
     }
+
+    return response;
+  }
+
+  private ResponseEntity<?> buildResponse(
```
Comment (igpetrov) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java:138]: This method seems to have too many branches. Consider maybe splitting it up? 🤔

**Review by igpetrov — APPROVED** [review]
LGTM

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java (hunk 1 above)
Comment (sbuettner) [reply connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java:138]: Good point. Pushed fix.

File: connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/CorrelationResult.java
Comment (chillleader) [inline connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/CorrelationResult.java:1]: Maybe we can move `MessageAlreadyCorrelated` to the `Success` sub-tree as discussed, wdyt?

**Review by chillleader — APPROVED** [review]

**Pull request merged**